### PR TITLE
Add rankings.v1.json report to the CreateAzureCdnWarehouseReports job

### PR DESCRIPTION
**Pull request by scottbommarito**
Added a rankings.v1.json report to the CreateAzureCdnWarehouseReports job.

We may want to put the query in a stored procedure on the database so we can remove that long script from the file.

(Fixes https://github.com/NuGet/NuGetGallery/issues/3134)

### CODE REVIEW

**Comment by maartenba** [comment]
Which parameters does this job need? Where does it write the file to?

@xavierdecoster would it make sense to have this in the generate auxiliary data job instead?

File: src/Stats.CreateAzureCdnWarehouseReports/Job.cs [hunk 1]
```diff
@@ -141,14 +141,23 @@ public override bool Init(IDictionary<string, string> jobArgsDictionary)
                 ApplicationInsightsHelper.TrackReportProcessed(DownloadCountReport.ReportName);
                 stopwatch.Restart();
 
+                // build rankings.v1.json
+                var rankingsReport = new RankingsReport(_cloudStorageAccount, _statisticsContainerName, _statisticsDatabase, _galleryDatabase);
```
Comment (maartenba) [inline src/Stats.CreateAzureCdnWarehouseReports/Job.cs]: Not sure the statistics container name here is correct, this should definitely go in another container (ng-search-data) in another storage account even.

File: src/Stats.CreateAzureCdnWarehouseReports/RankingsReport.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                        AND [Dimension_Date].[Date] < CONVERT(DATE, GETUTCDATE())
+                        AND (
+                                LOWER([Dimension_Operation].[Operation]) = 'install'
+                                OR
+                                LOWER([Dimension_Operation].[Operation]) = 'update'
+                            )
+
+                    GROUP BY    [Dimension_Package].[PackageId]
+                    ORDER BY    Downloads DESC";
+        internal const string ReportName = "rankings.v1.json";
```
Comment (maartenba) [inline src/Stats.CreateAzureCdnWarehouseReports/RankingsReport.cs]: Maybe add a newline before?

**Comment by xavierdecoster** [comment]
This report should be created as part of the **Search.GenerateAuxiliaryData** job. 

The CreateAzureCdnWarehouseReports job is one that is rather long running and takes a dependency on a cursor in the database to generate its output consistently, I don't think this blob should be part of that same cursor window. In fact, it is just auxiliary data for the search service, which is exactly the scope of the other job :)

**Comment by scottbommarito** [comment]
@maartenba 
The job requires `AzureCdnCloudStorageAccount`, `AzureCdnCloudStorageContainerName`,  `StatisticsDatabase`, `SourceDatabase`, `DataStorageAccount`, `InstrumentationKey`, `DataContainerName`.

File: src/Stats.CreateAzureCdnWarehouseReports/Job.cs (hunk 1 above)
Comment (scottbommarito) [inline src/Stats.CreateAzureCdnWarehouseReports/Job.cs]: Do you mean that it should go into `ng-search-data` or that it should go into a different container than `ng-search-data`? It actually already is outputting into `ng-search-data`, specifically `ng-search-data/rankings.v1.json`. Is that the correct place for it?

**Comment by scottbommarito** [comment]
@xavierdecoster I agree, I will move it.

**Comment by skofman1** [comment]
>                 stopwatch.Restart();

why restart here? It's the end of the method

---
Refers to: src/Stats.CreateAzureCdnWarehouseReports/Job.cs:169 in a39c34e. [](commit_id = a39c34eb6b891cecdf762d62fa52c99f77190c09, deletion_comment = False)

**Comment by scottbommarito** [comment]
@maartenba @xavierdecoster Would it be possible to send me a valid version of the `rankings.v1.json` file (over email perhaps) so I can get the formatting of it just right?

**Comment by skofman1** [comment]
>                 _logger.LogError("Job run failed!", exception);

please sync with latest version. I fixed those error logs format

---
Refers to: src/Stats.CreateAzureCdnWarehouseReports/Job.cs:175 in a39c34e. [](commit_id = a39c34eb6b891cecdf762d62fa52c99f77190c09, deletion_comment = False)

**Comment by scottbommarito** [comment]
Closing this because adding rankings.v1.json to GenerateAuxiliaryData with a new branch was easier than adding more commits to this one.

**Pull request closed without merging**